src/main/java/org/zone/region/flag/entity/player/display/title/TitleMessageDisplay.java

```diff
@@ -3,40 +3,121 @@
 import org.zone.region.flag.entity.player.display.MessageDisplay;
 import org.zone.region.flag.entity.player.display.MessageDisplayType;
 import org.zone.region.flag.entity.player.display.MessageDisplayTypes;
 
+import java.time.Duration;
+import java.util.Optional;
+
 public class TitleMessageDisplay implements MessageDisplay {
 
     private @NotNull Component subTitle;
-    private @NotNull Title.Times times;
+    private @Nullable Duration fadeIn;
+    private @Nullable Duration stay;
+    private @Nullable Duration fadeOut;
 
-    public TitleMessageDisplay(@NotNull Component subTitle, @NotNull Title.Times times) {
-        this.subTitle = subTitle;
-        this.times = times;
+    public TitleMessageDisplay(@NotNull TitleMessageDisplayBuilder builder) {
+        this.subTitle = builder.getSubTitle();
+        this.fadeIn = builder.getFadeIn();
+        this.stay = builder.getStay();
+        this.fadeOut = builder.getFadeOut();
     }
 
+    /**
+     * Gets the subtitle of the Title Message
+     *
+     * @return The subtitle
+     */
     public @NotNull Component getSubTitle() {
         return this.subTitle;
     }
 
-    public @NotNull Title.Times getTimes() {
-        return this.times;
+    /**
+     * Gets the fade in duration of the Title message
+     *
+     * @return Optional nullable version of fade in duration
+     */
+    public @NotNull Optional<Duration> getFadeIn() {
+        return Optional.ofNullable(this.fadeIn);
+    }
+
+    /**
+     * Gets the stay duration of the Title message
+     *
+     * @return Optional nullable version of stay duration
+     */
+    public @NotNull Optional<Duration> getStay() {
+        return Optional.ofNullable(this.stay);
+    }
+
+    /**
+     * Gets the fade out duration of the Title message
+     *
+     * @return Optional nullable version of fade out duration
+     */
+    public @NotNull Optional<Duration> getFadeOut() {
+        return Optional.ofNullable(this.fadeOut);
     }
 
+    /**
+     * Sets the subtitle of the Title message
+     *
+     * @param subTitle The subtitle component to be set
+     */
     public void setSubTitle(@NotNull Component subTitle) {
         this.subTitle = subTitle;
     }
 
-    public void setTimes(@NotNull Title.Times times) {
-        this.times = times;
+    /**
+     * Sets the fade in duration of the Title message
+     *
+     * @param fadeIn The fade in duration to set
+     */
+    public void setFadeIn(@NotNull Duration fadeIn) {
+        this.fadeIn = fadeIn;
+    }
+
+    /**
+     * Sets the stay duration of the Title message
+     *
+     * @param stay The stay duration to set
+     */
+    public void setStay(@NotNull Duration stay) {
+        this.stay = stay;
+    }
+
+    /**
+     * Sets the fade out duration of the Title message
+     *
+     * @param fadeOut The fade out duration to set
+     */
+    public void setFadeOut(@NotNull Duration fadeOut) {
+        this.fadeOut = fadeOut;
```

Comment:
FadeIn, FadeOut, Stay are all nullable in the fields, why make them ``@NotNull`` here?

Reply:
well i didn't want the method to be used to set `null` to `fadeOut`

Reply:
Fair enough. The issue is someone may want to change it to the default and instead of just setting this to null, they need to emulate what your already doing. 

I personally am with you, however i set it as nullable and then have another method that is preferred for setting it to default

Reply:
hmm..